src/darker/formatters/black_formatter.py

```diff
@@ -0,0 +1,222 @@
+    magic_trailing_comma: bool
+    preview: bool
+
+
+class BlackFormatter(BaseFormatter):
+    """Black code formatter plugin interface."""
+
+    def __init__(self) -> None:  # pylint: disable=super-init-not-called
+        """Initialize the Black code re-formatter plugin."""
+        self.config: BlackConfig = {}
+
+    name = "black"
+
+    def read_config(self, src: tuple[str, ...], args: Namespace) -> None:
+        """Read Black configuration from ``pyproject.toml``.
+
+        :param src: The source code files and directories to be processed by Darker
+        :param args: Command line arguments
+
+        """
+        config_path = args.config or find_pyproject_toml(src)
+        if config_path:
+            self._read_config_file(config_path)
+        self._read_cli_args(args)
+
+    def _read_config_file(self, config_path: str) -> None:  # noqa: C901
+        raw_config = parse_pyproject_toml(config_path)
+        if "line_length" in raw_config:
+            self.config["line_length"] = raw_config["line_length"]
+        if "skip_magic_trailing_comma" in raw_config:
+            self.config["skip_magic_trailing_comma"] = raw_config[
+                "skip_magic_trailing_comma"
+            ]
+        if "skip_string_normalization" in raw_config:
+            self.config["skip_string_normalization"] = raw_config[
+                "skip_string_normalization"
+            ]
+        if "preview" in raw_config:
+            self.config["preview"] = raw_config["preview"]
+        if "target_version" in raw_config:
+            target_version = raw_config["target_version"]
+            if isinstance(target_version, str):
+                self.config["target_version"] = target_version
+            elif isinstance(target_version, list):
+                # Convert TOML list to a Python set
+                self.config["target_version"] = set(target_version)
+            else:
+                message = (
+                    f"Invalid target-version = {target_version!r} in {config_path}"
+                )
+                raise ConfigurationError(message)
+        if "exclude" in raw_config:
+            self.config["exclude"] = re_compile_maybe_verbose(raw_config["exclude"])
+        if "extend_exclude" in raw_config:
+            self.config["extend_exclude"] = re_compile_maybe_verbose(
+                raw_config["extend_exclude"]
+            )
+        if "force_exclude" in raw_config:
+            self.config["force_exclude"] = re_compile_maybe_verbose(
+                raw_config["force_exclude"]
+            )
+
+    def _read_cli_args(self, args: Namespace) -> None:
+        if args.config:
+            self.config["config"] = args.config
+        if getattr(args, "line_length", None):
+            self.config["line_length"] = args.line_length
+        if getattr(args, "target_version", None):
+            self.config["target_version"] = {args.target_version}
+        if getattr(args, "skip_string_normalization", None) is not None:
```

Comment:
nit: can remove is not None and just do if as used in the above

Reply:
But Darker has the `--no-skip-string-normalization` option, and if it's specified, we want to add `skip_string_normalization=False` to the configuration.